src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,210 @@
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, colony, raidingWorld);
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final Colony colony, final World world)
+    {
+
+        final BlockPos targetSpawnPoint = calculateSpawnLocation(world, colony);
+
+        if (targetSpawnPoint != null && entityToSpawn != null && world != null)
+        {
+            final int x = targetSpawnPoint.getX();
+            final int y = targetSpawnPoint.getY();
+            final int z = targetSpawnPoint.getZ();
+
+            IntStream.range(0, numberOfSpawns).forEach(theInteger ->
+            {
+                final Entity entity = EntityList.createEntityByIDFromName(entityToSpawn, world);
+
+                if (entity != null)
+                {
+                    if (entityToSpawn.equals(barbarian))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.STONE_AXE));
+                    }
+                    if (entityToSpawn.equals(archer))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.BOW));
+                    }
+                    if (entityToSpawn.equals(chief))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.GOLDEN_SWORD));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.CHEST, new ItemStack(Items.CHAINMAIL_CHESTPLATE));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.LEGS, new ItemStack(Items.CHAINMAIL_LEGGINGS));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.FEET, new ItemStack(Items.CHAINMAIL_BOOTS));
+                    }
+                    entity.setLocationAndAngles(x, y, z, MathHelper.wrapDegrees(world.rand.nextFloat() * WHOLE_CIRCLE), 0.0F);
+                    world.spawnEntity(entity);
+                }
+            });
+        }
+    }
+
+    /**
+     * Calculate the colony border.
+     *
+     * @param theWorld in the world.
+     * @param colony   the Colony to spawn the barbarians near.
+     */
+    private static BlockPos calculateSpawnLocation(final World theWorld, final Colony colony)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(theWorld, colony.getCenter());
+        if (colonyView == null)
+        {
+            return null;
+        }
+        final BlockPos center = colonyView.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
+
+        final int RandomDegree = theWorld.rand.nextInt(NUMBER_OF_POSSIBLE_CASES);
+
+        for (double degrees = 0; degrees < (int) WHOLE_CIRCLE; degrees += 1)
+        {
+            if (degrees == RandomDegree)
```

Comment:
yeah, you don't need the for loop, just put in the if with the right values already =D